recipes-kernel/linux/files/patches-5.10/0119-arm64-dts-ti-iot2050-add-the-support-for-the-m.2-var.patch

```diff
@@ -0,0 +1,311 @@
+@@ -0,0 +1,29 @@
++// SPDX-License-Identifier: GPL-2.0
++/*
++ * DT overlay for M2 card:B-KEY PCIE0_LANE0 E-KEY PCIE1_LANE0
++ * Copyright (c) Siemens AG, 2022
++ *
++ * Authors:
++ *   Chao Zeng <chao.zeng@siemens.com>
++ */
++
++/dts-v1/;
++/plugin/;
++#include <dt-bindings/phy/phy.h>
++#include <dt-bindings/phy/phy-am654-serdes.h>
++#include <dt-bindings/gpio/gpio.h>
++
++&pcie1_rc {
++	num-lanes = <1>;
++	phys = <&serdes1 PHY_TYPE_PCIE 0>;
++	phy-names = "pcie-phy0";
++	status = "okay";
++};
++
++&pcie0_rc {
++	num-lanes = <1>;
++	phys = <&serdes0 PHY_TYPE_PCIE 1>;
++	phy-names = "pcie-phy0";
++	reset-gpios = <&main_gpio1 15 GPIO_ACTIVE_HIGH>;
++	status = "okay";
++};
+diff --git a/arch/arm64/boot/dts/ti/k3-am654-iot2050-bkey-pciex2.dts b/arch/arm64/boot/dts/ti/k3-am654-iot2050-bkey-pciex2.dts
+new file mode 100644
+index 000000000000..65515069898d
+--- /dev/null
++++ b/arch/arm64/boot/dts/ti/k3-am654-iot2050-bkey-pciex2.dts
+@@ -0,0 +1,29 @@
++// SPDX-License-Identifier: GPL-2.0
++/*
++ * DT overlay for M2 card:B-KEY PCIE0_LANE0 and PCIE0_LANE1 *
++ * Copyright (c) Siemens AG, 2022
++ *
++ * Authors:
++ *   Chao Zeng <chao.zeng@siemens.com>
++ */
++
++/dts-v1/;
++/plugin/;
++#include <dt-bindings/phy/phy.h>
++#include <dt-bindings/phy/phy-am654-serdes.h>
++#include <dt-bindings/gpio/gpio.h>
++
++&pcie0_rc {
++	pinctrl-names = "default";
++	pinctrl-0 = <&main_bkey_pcie_reset>;
++
++	num-lanes = <2>;
++	phys = <&serdes0 PHY_TYPE_PCIE 1>, <&serdes1 PHY_TYPE_PCIE 1>;
++	phy-names = "pcie-phy0","pcie-phy1";
++	reset-gpios = <&main_gpio1 15 GPIO_ACTIVE_HIGH>;
++	status = "okay";
++};
++
++&pcie1_rc {
++	status = "disabled";
++};
+diff --git a/arch/arm64/boot/dts/ti/k3-am654-iot2050-bkey-usb3-ekey-pcie.dts b/arch/arm64/boot/dts/ti/k3-am654-iot2050-bkey-usb3-ekey-pcie.dts
+new file mode 100644
+index 000000000000..922a2eef4c55
+--- /dev/null
++++ b/arch/arm64/boot/dts/ti/k3-am654-iot2050-bkey-usb3-ekey-pcie.dts
+@@ -0,0 +1,32 @@
++// SPDX-License-Identifier: GPL-2.0
++/*
++ * DT overlay for M2 card:B-KEY USB3.0 E-KEY PCIE1_LANE0 *
++ * Copyright (c) Siemens AG, 2022
++ *
++ * Authors:
++ *   Chao Zeng <chao.zeng@siemens.com>
++ */
++
++/dts-v1/;
++/plugin/;
++#include <dt-bindings/phy/phy.h>
++#include <dt-bindings/phy/phy-am654-serdes.h>
++
++&pcie1_rc {
++	num-lanes = <1>;
++	phys = <&serdes1 PHY_TYPE_PCIE 0>;
++	phy-names = "pcie-phy0";
++};
++
++&dwc3_0 {
++	assigned-clock-parents = <&k3_clks 151 4>,  /* set REF_CLK to 20MHz i.e. PER0_PLL/48 */
++				 <&k3_clks 151 8>;  /* set PIPE3_TXB_CLK to WIZ8B2M4VSB */
++	phys = <&serdes0 PHY_TYPE_USB3 0>;
++	phy-names = "usb3-phy";
++};
++
++&usb0 {
++	maximum-speed = "super-speed";
++	snps,dis-u1-entry-quirk;
++	snps,dis-u2-entry-quirk;
++};
+diff --git a/arch/arm64/boot/dts/ti/k3-am6548-iot2050-advanced-m2.dts b/arch/arm64/boot/dts/ti/k3-am6548-iot2050-advanced-m2.dts
+new file mode 100644
+index 000000000000..32cc989e88e6
+--- /dev/null
++++ b/arch/arm64/boot/dts/ti/k3-am6548-iot2050-advanced-m2.dts
```

Comment:
Also here the naming question: "advanced-m2" suggests that there could be "basic-m2" as well. Is there?

Reply:
I don't know, but I think we can provides more scalability opportunities.
anyway, we should re-consider the naming conventions